### PR TITLE
NEON 51 - Latitude longitude aggregation

### PR DESCRIPTION
**Pull request by jacklechnercaci**
Fixes: [NEON-51](https://nextcentury.atlassian.net/browse/NEON-51)

-Latlon clustering is now functional
-Two test cases have been provided, one with secondary keys and one without

### CODE REVIEW

File: server/src/main/java/com/ncc/neon/services/ClusterService.java [hunk 1]
```diff
@@ -208,6 +224,62 @@ public TabularQueryResult cluster(TabularQueryResult tabularQueryResult) {
...
+     */
+    private List<Map<String, Object>> getNewLatLonDataBins(List<String> fieldNames, BigDecimal longCount, List<List<Object>> clusters) {
+        List<Map<String, Object>> newData = new ArrayList<>();
+
+        String latKey = fieldNames.get(0);
+        String longKey = fieldNames.get(1);
+
+        BigDecimal latCount = longCount.divide(new BigDecimal(2));
+        //TODO: round this?
+        BigDecimal totalCount = longCount.multiply(latCount);
```
Comment (ThomasSchellenbergNextCentury) [inline server/src/main/java/com/ncc/neon/services/ClusterService.java:243]: Both counts should be integers, right?  So I don't think there's a need to round them.

File: server/src/main/java/com/ncc/neon/services/ClusterService.java [hunk 2]
```diff
@@ -208,6 +224,62 @@ public TabularQueryResult cluster(TabularQueryResult tabularQueryResult) {
...
+
+        BigDecimal step = new BigDecimal(DEFAULT_LATLON_STEP);
+
+        BigDecimal longGap = new BigDecimal(360).divide(longCount);
+        BigDecimal latGap = new BigDecimal(180).divide(latCount);
+
+        if (clusters == null) {
+            // lat first
+            System.out.println();
```
Comment (ThomasSchellenbergNextCentury) [inline server/src/main/java/com/ncc/neon/services/ClusterService.java:252]: Remove the println

File: server/src/main/java/com/ncc/neon/services/ClusterService.java (hunk 1 above)
Comment (jacklechnercaci) [reply server/src/main/java/com/ncc/neon/services/ClusterService.java:243]: I actually forgot this was in here!

File: server/src/main/java/com/ncc/neon/services/ClusterService.java (hunk 2 above)
Comment (jacklechnercaci) [reply server/src/main/java/com/ncc/neon/services/ClusterService.java:252]: Oops, trusty break spot. I will remove it.

Comment (jacklechnercaci) [reply server/src/main/java/com/ncc/neon/services/ClusterService.java]: Fixed.

File: server/src/main/java/com/ncc/neon/services/ClusterService.java (hunk 1 above)
Comment (jacklechnercaci) [reply server/src/main/java/com/ncc/neon/services/ClusterService.java]: Fixed.

**Pull request merged**